Use MySQL < v5.7.22 in tests

We have had multiple issues with using features in MySQL that are not supported in all deployments (See #5350). Changing the MySQL version in tests caused the following test failure, and prevented making the change to tests. We should resolve this.

```
$ make test-go
...
--- FAIL: TestNewUsesRegisterTLS (0.07s)
    mysql_test.go:781: 
        	Error Trace:	mysql_test.go:781
        	Error:      	Not equal: 
        	            	expected: "x509: certificate is not valid for any names, but wanted to match localhost"
        	            	actual  : "tls: server selected unsupported protocol version 302"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-x509: certificate is not valid for any names, but wanted to match localhost
        	            	+tls: server selected unsupported protocol version 302
        	Test:       	TestNewUsesRegisterTLS
```

--- a/server/datastore/mysql/mysql_test.go
+++ b/server/datastore/mysql/mysql_test.go
@@ -778,7 +778,9 @@ func TestNewUsesRegisterTLS(t *testing.T) {
 	// This fails because the certificate mysql is using is different than the one generated here
 	_, err := newDSWithConfig(t, dbName, mysqlConfig)
 	require.Error(t, err)
-	require.Equal(t, "x509: certificate is not valid for any names, but wanted to match localhost", err.Error())
+	// TODO: we're using a Regexp because the message is different depending on the version of mysql,
+	// we should refactor and use different error types instead.
+	require.Regexp(t, "^(x509|tls)", err.Error())
 }
 
 func TestWhereFilterTeams(t *testing.T) {

